src/main/java/com/artipie/management/api/ApiRepoDeleteSlice.java

```diff
@@ -0,0 +1,161 @@
+                        final Key repo = new Key.From(user, String.format("%s.yaml", name));
+                        return this.configfile.exists(repo)
+                            .thenCompose(
+                                exists -> {
+                                    final CompletionStage<Void> res;
+                                    if (exists) {
+                                        res = this.configfile.delete(repo)
+                                            .thenCompose(
+                                                noth -> this.storage.list(
+                                                    new Key.From(user, name)
+                                                ).thenCompose(
+                                                    keys -> CompletableFuture.allOf(
+                                                        keys.stream().map(
+                                                            this.storage::delete
+                                                        ).toArray(CompletableFuture[]::new)
+                                                    )
+                                                )
+                                            );
+                                    } else {
+                                        res = new FailedCompletionStage<>(
+                                            new ArtipieException(
+                                                String.format(
+                                                    "Repo '%s' does not exist", repo.string()
+                                                )
+                                            )
+                                        );
+                                    }
+                                    return res;
+                                }
+                            ).thenApply(nothing -> repo);
+                    }).handle(
+                        (repo, thr) -> {
+                            final Response res;
+                            if (thr == null) {
+                                res = new RsWithHeaders(
+                                    new RsWithStatus(RsStatus.OK),
+                                    new Headers.From(
+                                        "Location", String.format("/dashboard/%s", user)
+                                    )
+                                );
+                            } else if (thr.getCause() instanceof ArtipieException) {
+                                res = new RsWithBody(
+                                    new RsWithStatus(RsStatus.BAD_REQUEST),
+                                    String.format("Failed to delete repo '%s'", repo),
+                                    StandardCharsets.UTF_8
+                                );
+                            } else {
+                                res = new RsWithStatus(RsStatus.INTERNAL_ERROR);
+                            }
+                            return res;
```

Comment:
hm, cannot we get rid of this `handle` by returning `CompletionStage<Response>` from previous `thenCompose`? In the case of some exception it will be handled on the upper level and `INTERNAL_ERROR` will be returned, will not it?